scripts/prepare_disk.sh

```diff
@@ -22,6 +22,58 @@ usage()
     echo "NOTE: This script is destructive and will destroy your data."
 }
 
+is_integer()
+{
+    test "${1}" -eq "${1}" 2> /dev/null
+}
+
+is_comment()
+{
+    test -z "${1%%#*}"
+}
+
+# A bit counter intuitive, a function can either return an exit code,
+# or set a global variable, or echo something, in this case the
+# exit code is 0 if a resize is required.
+is_resize_needed()
+{
+    current_partition_table_file="$(mktemp)"
+    sfdisk -d "${TARGET_DISK}" > "${current_partition_table_file}"
+
+    changes_needed=false
+    while IFS="${IFS}:=," read -r table_label _ table_start _ table_size _; do
+        if is_comment "${table_label}" || ! is_integer "${table_start}" || \
+            ! is_integer "${table_size}"; then
+            continue
+        fi
+
+        while IFS="${IFS}:=," read -r disk_label _ disk_start _ disk_size _; do
+            if is_comment "${disk_label}" || ! is_integer "${disk_start}" || \
+                ! is_integer "${disk_size}"; then
+                continue
+            fi
+
+            if [ "${table_label}" != "${disk_label}" ]; then
+                continue
+            fi
+
+            if [ "${table_start}" -ne "${disk_start}" ] || \
+               [ "${table_size}" -ne "${disk_size}" ]; then
+                changes_needed=true
+                return # returns from both loops!
```

Comment:
Shouldn't this be "return 0" to indicate that changes are needed? As as far as I can see this returns from both loops and the function. And thus setting changes_needed has no real effect within the function.

Reply:
You'd have to double check with raymond, but he found it did not. return only exists the subshell the while loop is being run under, as such, return should just return whatever was the last exit value within this subshell.

Reply:
Although it works, it does not look nice, i definitely agree. Nonetheless i decided to validate my finding once more it seems that i put myself on the wrong track before, probably with the 'sfdisk | while ...' construct, triggering an extra subshell. Much better now :P